internal_packages/keybase/lib/preferences-keybase.cjsx

```diff
@@ -0,0 +1,242 @@
+      selectedPrivKey = if privKeys.length > 0 then privKeys[0] else null
+
+    return {pubKeys, privKeys, selectedPubKey, selectedPrivKey}
+
+  componentDidMount: ->
+    @usub = PGPKeyStore.listen @_onChange
+
+  ###
+  componentWillUnmount: ->
+    @usub()
+    if @state.selectedKey?
+      @_saveKeyNow(@state.selectedKey.name)
+  ###
+
+  # Key Editing
+
+  _onSelectPubKey: (event) =>
+    selectedPubKey = null
+    for pubKey in @state.pubKeys
+      if event.target.value in privKey.addresses
+        selectedPubKey = pubKey
+    @setState
+      selectedPubKey: selectedPubKey
+
+  _onSelectPrivKey: (event) =>
+    selectedPrivKey = null
+    for privKey in @state.privKeys
+      if event.target.value in privKey.addresses
+        selectedPrivKey = privKey
+    @setState
+      selectedPrivKey: selectedPrivKey
+
+  # DELETE AND NEW
+  _deletePubKey: =>
+    numKeys = @state.pubKeys.length
+    if @state.selectedPubKey?
+      PGPKeyStore.deleteKey(@state.selectedPubKey)
+    if numKeys==1
+      @setState
+        editState: "new"
+        selectedPubKey: null
+
+  _deletePrivKey: =>
+    numKeys = @state.privKeys.length
+    if @state.selectedPrivKey?
+      PGPKeyStore.deleteKey(@state.selectedPrivKey)
+    if numKeys==1
+      @setState
+        editState: "new"
+        selectedPrivKey: null
+
+  _startNewPubKey: =>
+    @setState
+      editState: "new"
+      selectedPubKey: null
+
+  _startNewPrivKey: =>
+    @setState
+      editState: "new"
+      selectedPrivKey: null
+
+  _saveNewPubKey: =>
+    PGPKeyStore.saveNewKey("benbitdiddle@icloud.com", TEST_KEY, isPub = true)
+
+  _saveNewPrivKey: =>
+    PGPKeyStore.saveNewKey("benbitdiddle@icloud.com", TEST_KEY, isPub = false)
+
+  _addOtherAddress: =>
+    PGPKeyStore.addAddressToKey(@state.pubKeys[0], "logan@nylas.com")
+
+  _removeOtherAddress: =>
+    PGPKeyStore.removeAddressFromKey(@state.pubKeys[0], "logan@nylas.com")
+
+  ###
+  _renderCreateNew: ->
+    cancel = <button className="btn key-name-btn" onClick={@_cancelNewKey}>Cancel</button>
+    <div className="section-title">
+      Key Name: <input type="text" className="key-name-input" value={@state.selectedKeyName} onChange={@_onEditName}/>
+      <button className="btn btn-emphasis key-name-btn" onClick={@_saveNewKey}>Save</button>
+      {if @state.pubKeys.length then cancel}
+    </div>
+  ###
+
+  _renderPublicKeyPicker: ->
+    options = @state.pubKeys.map (pubKey) ->
+      <option value={pubKey.addresses[0]} key={pubKey.addresses[0]}>{pubKey.addresses[0]}</option>
+
+    <select value={@state.selectedPubKey?.addresses[0]} onChange={@_onSelectPubKey}>
+      {options}
+    </select>
+
+  _renderPrivateKeyPicker: ->
+    options = @state.privKeys.map (privKey) ->
+      <option value={privKey.addresses[0]} key={privKey.addresses[0]}>{privKey.addresses[0]}</option>
+
+    <select value={@state.selectedPrivKey?.addresses[0]} onChange={@_onSelectPrivKey}>
+      {options}
+    </select>
+
+  _renderName: ->
```

Comment:
There used to be a `KeyManager` and `KeybaseSearch` component, but the problem was that there's some bleedover - so, for instance, searching keybase could turn up someone you have a key saved for, and it was unclear how to handle that in two entirely different components.